### DISCUSSION TITLE
Facing issues while upgrading version for my project.

### DISCUSSION THREAD

**Discussion by vaibhavmehra9** [react-hook-form/react-hook-form#8002] [Q&A] [answered]
Hi, everyone my project is currently using the version 7.0.1 and I am trying to upgrade it into latest version but facing some issues due to deprecated changes. In my project somewhere **control.defaultValuesRef.current** is using, now it is deprecated in the latest version. please help if anyone faced the same problem and how to resolved it

**Comment by Moshyfawn** [comment, marked as answer]
It's generally unadvised to use `control` other than to pass it into the RHF methods / hooks.
From [RHF docs on `control`](https://react-hook-form.com/api/useform/control):
> Important: do not access any of the properties inside this object directly. It's for internal usage only.

You'll need to refactor your project to pass your `defaultValues` object down the React tree instead.
